### PR TITLE
Document new CLI commands and add Local Dev page

### LINKED ISSUE #386: New CLI commands for dev
I've recently added many commands that are useful for development. I think we should have a sort of _cheatsheet_ of commands useful for development. Where and how IDK :smile: 

## Cancel batch

```
restate invocation cancel
````

Now accepts `service/handler` or `vObj/key/handler` etc etc. See https://github.com/restatedev/restate/blob/main/cli/src/commands/invocations/cancel.rs#L25

## Prune (batch too)

To remove an invocation in **completed** state:

```
restate invocation prune
```

Accepts a target string similarly to command above. See https://github.com/restatedev/restate/blob/main/cli/src/commands/invocations/purge.rs#L25

## Clear all kv state

Clear all KV state of virtual objects/workflows

```
restate kv clear 
```

Accepts either `serviceName/serviceKey` or only `serviceName`. See https://github.com/restatedev/restate/blob/main/cli/src/commands/state/clear.rs#L27

## SQL command with output piping

The sql command is not hidden anymore, and it can output json:

```
restate sql --json "query"
```

This is very useful for piping commands with jq and do some useful admin stuff.

### PR DESCRIPTION
**Pull request by gvdongen**
-Fixes #386 

- Created a local dev page and moved the CLI installation there. 
- Added the installation of the Restate server options as well 
- Moved the remainder of the previous CLI page to the configuration section
- Made the quickstarts more concise and let them refer to the local dev page

### CODE REVIEW

File: docs/develop/local_dev.mdx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    ```
+
+</CH.Spotlight>
+<CH.Spotlight className={"spotlight-medium"}>
+
+    Execute a SQL query on the invocation or application state.
+    See [SQL introspection docs](/operate/introspection?interface=psql#inspecting-invocations) for example queries.
+
+    ```shell
+    restate sql --json "query"
```
Comment (slinkydeveloper) [inline docs/develop/local_dev.mdx:171]: Remove the `--json` example here, if you don't need to pipe the `--json` output is hard to read.

Perhaps you can briefly mention an example where you show how to pipe `--json` output to some other command using jq, for example:

```
# Cancel all pending invocations to service "Greeter"
restate sql --json "SELECT id FROM sys_invocation WHERE target_service_name = 'Greeter' AND status = 'pending'" | jq .id | xargs restate invocation cancel --yes
```

File: docs/develop/local_dev.mdx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    restate-server
+    ```
+</CH.Spotlight>
+<CH.Spotlight className={"spotlight-medium"}>
+    Start the Restate Server and wipe ongoing invocations and K/V state.
+
+    ```shell
+    restate-server --wipe all
+    ```
+</CH.Spotlight>
```
[the remark this answers is not in the document]
Comment (gvdongen) [reply docs/develop/local_dev.mdx:107]: Hmm, I disagree, for local development this is still a useful command to know about... The commands to get you unstuck are below. And these docs are more like reference docs, so I feel this should be in there.

[the remark this answers is not in the document]
Comment (gvdongen) [reply docs/develop/local_dev.mdx:107]: To give an example, I run one example, and then want to run another. I don't want to go through cancelling and clearing everything I did, I just want to start anew.

[the remark this answers is not in the document]
Comment (slinkydeveloper) [reply docs/develop/local_dev.mdx:107]: > so I feel this should be in there.

Fair enough, but still I would prefer if this is not at the beginning, as the second command you read about. We got the feedback more than once that this command makes restate feel like it doesn't give you the tools to unblock yourself.

File: docs/develop/local_dev.mdx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        Install the Restate Server and CLI by downloading the binaries with `curl` from the [releases page](https://github.com/restatedev/restate/releases/latest), and make them executable:
+
+        <CH.Code rows={"5"}>
+
+            ```shell MacOS-x64
+            curl -LO https://github.com/restatedev/restate/releases/latest/download/restate.x86_64-apple-darwin.tar.gz && \
+            tar -xvf restate.x86_64-apple-darwin.tar.gz && \
+            chmod +x restate restate-server && \
+            sudo mv restate /usr/local/bin/ && \
+            sudo mv restate-server /usr/local/bin/
```
Comment (slinkydeveloper) [inline docs/develop/local_dev.mdx:64]: Add a `BIN` variable to this and below command, so it's easier to change where you install it. I don't like we have commands with `sudo` in it.

```shell
BIN=/usr/local/bin && \
curl -LO https://github.com/restatedev/restate/releases/latest/download/restate.x86_64-apple-darwin.tar.gz && \
            tar -xvf restate.x86_64-apple-darwin.tar.gz && \
            chmod +x restate restate-server && \
            mv restate $BIN && \
            mv restate-server $BIN
```

File: docs/develop/local_dev.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (gvdongen) [reply docs/develop/local_dev.mdx]: Fair enough but this is not the second thing you read. This is part of the reference docs so you should see those as an unstructured collection of SDK docs. 
Users will start with the quickstart, concepts, examples and tour.

**Pull request merged**